sr_utilities/test/test_hand_finder.cpp

```diff
@@ -54,6 +121,134 @@ TEST(SrHandFinder, hand_present_test)
   {
     ROS_INFO_STREAM(iter->second);
+    ASSERT_STREQ(iter->second.c_str(), (ethercat_path + "/calibrations/1/" + "calibration.yaml").c_str());
+  }
+  shadow_robot::HandControllerTuning controller_tuning(hand_finder.get_hand_controller_tuning());
+  for (map<string, string>::const_iterator iter = controller_tuning.friction_compensation_.begin();
+       iter != controller_tuning.friction_compensation_.end(); ++iter)
+  {
+    ASSERT_STREQ(iter->second.c_str(), (ethercat_path + "/controls/friction_compensation.yaml").c_str());
+    ROS_DEBUG_STREAM(iter->second);
+  }
+  for (map<string, string>::const_iterator iter = controller_tuning.motor_control_.begin();
+       iter != controller_tuning.motor_control_.end(); ++iter)
+  {
+    ASSERT_STREQ(iter->second.c_str(),
+                 (ethercat_path + "/controls/motors/1/motor_board_effort_controllers.yaml").c_str());
+    ROS_DEBUG_STREAM(iter->second);
+  }
+  for (map<string, vector<string> >::const_iterator iter = controller_tuning.host_control_.begin();
+       iter != controller_tuning.host_control_.end(); ++iter)
+  {
+    string host_array[] = {"sr_edc_calibration_controllers.yaml",
+                           "sr_edc_joint_velocity_controllers_PWM.yaml",
+                           "sr_edc_effort_controllers_PWM.yaml",
+                           "sr_edc_joint_velocity_controllers.yaml",
+                           "sr_edc_effort_controllers.yaml",
+                           "sr_edc_mixed_position_velocity_joint_controllers_PWM.yaml",
+                           "sr_edc_joint_position_controllers_PWM.yaml",
+                           "sr_edc_mixed_position_velocity_joint_controllers.yaml",
+                           "sr_edc_joint_position_controllers.yaml"};
+    vector<string> host_controller_files(host_array, host_array + 9);
+    ASSERT_EQ(host_controller_files.size(), iter->second.size());
+    for (size_t i = 0; i != iter->second.size(); ++i)
+    {
+      ASSERT_STREQ(iter->second[i].c_str(), (ethercat_path + "/controls/host/1/" + host_controller_files[i]).c_str());
+      ROS_DEBUG_STREAM(iter->second[i]);
+    }
+  }
+  nh.deleteParam("hand/mapping/1");
+  nh.deleteParam("hand/joint_prefix/1");
+  ASSERT_TRUE(true);
+}
+
+TEST(SrHandFinder, hand_nomapping_noprefix_test)
+{
+  ros::NodeHandle nh;
+  nh.setParam("hand/mapping/1", "");
+  nh.setParam("hand/joint_prefix/1", "");
+  const string joint_names[] = {"FFJ1", "FFJ2", "FFJ3", "FFJ4", "MFJ1", "MFJ2", "MFJ3", "MFJ4",
+                                "RFJ1", "RFJ2", "RFJ3", "RFJ4", "LFJ1", "LFJ2", "LFJ3", "LFJ4", "LFJ5",
+                                "THJ1", "THJ2", "THJ3", "THJ4", "THJ5", "WRJ1", "WRJ2"};
+  shadow_robot::SrHandFinder hand_finder;
+  map<string, vector<string> > hand_joints(hand_finder.get_joints());
+  for (map<string, vector<string> >::const_iterator iter = hand_joints.begin(); iter != hand_joints.end(); ++iter)
```

Comment:
I think that in this case `hand_joints` will be an empty map, isn't it? So maybe we should check:

```cpp
 ASSERT_TRUE(hand_joints.empy());
```